src/compiler/parser.ts

```diff
@@ -5698,9 +5700,10 @@ namespace ts {
             const hasJSDoc = hasPrecedingJSDocComment();
             parseExpected(SyntaxKind.WhileKeyword);
             const openParenPosition = scanner.getTokenPos();
-            parseExpected(SyntaxKind.OpenParenToken);
+            const openParenExists = parseExpected(SyntaxKind.OpenParenToken);
             const expression = allowInAnd(parseExpression);
-            parseExpectedMatchingBrackets(SyntaxKind.OpenParenToken, SyntaxKind.CloseParenToken, openParenPosition);
+            openParenExists ? parseExpectedMatchingBrackets(SyntaxKind.OpenParenToken, SyntaxKind.CloseParenToken, openParenPosition)
+            : parseExpected(SyntaxKind.CloseParenToken);

```

Comment:
This is needed in parseObjectLiteralExpression?

I think instead of changing this in all use cases, why not modify `parseExpectedMatchingBrackets` to take in result of openingBracket so all cases are always covered and we dont need to review all use cases...

Reply:
I don't think it should be needed for object literal expression (or arrays) since we only try to parse them if an opening bracket has already been parsed (or scanned, not sure what the correct terminology is there).